### DISCUSSION TITLE
Issue in Fetching Data from tables: NextJs14 with JS

### DISCUSSION THREAD

**Discussion by greeenboi** [supabase/supabase#20029] [Questions]
This seems to be a recurring issue in my project.

Below is the code that I use to fetch data from my table

```js
useEffect(() => {
    const fetchPosts = async () => {
      let { data: posts, error } = await supabase
        .from('posts')
        .select()
      console.log(posts)
      if(error) console.log(error) 
      else setBlogPosts(posts) && console.log(posts)
    }
    fetchPosts()
  
  }, [])
```

the response that i receive on the console is always an empty array.

I have:
- tried disabling rls 
- tried different browsers
- tried the same on a test ViteJs project 
- added policies to allow read and write to the public

I want to know how i can fix this issue.
The [Repository](https://github.com/greeenboi/sample_blog/tree/master) if you want to see it for reference.

**Comment by GaryAustin1** [comment]
Probably caching.
If you change .select() to .select('*') do you get data with RLS disabled?
https://github.com/orgs/supabase/discussions/19084
